Add explanation for StripPrefix behavior (#2888)
pivotal/docs-tap: api-auto-registration/key-concepts.hbs.md

  curated API
- `routeConfig` section specifies service level configuration you add when generating the routing
  resource for the API. For information about spring-cloud-gateway fields, see [OpenAPI route conversion](https://docs.vmware.com/en/VMware-Spring-Cloud-Gateway-for-Kubernetes/2.0/scg-k8s/GUID-guides-openapi-route-conversion.html).
  - `routeConfig.filters` section specifies service level filters for all the routes exposed in each

>     API. You may add modifications to your endpoints such as `RateLimit=5,10s` or `RemoveRequestHeader=X-Request-Foo`

Reviewer: Should we add a link to the filters and predicates page ?

  - Your controller will automatically prepend each endpoint path with the `pathPrefix` you specified
    for each APIDescriptor, and add `StripPrefix` filter to the end of the filter list to facilitate

>     a successful path-based redirect. But do not worry, you may still add your own `StripPrefix` filter

Reviewer: ```suggestion
    a successful path-based redirect. Additionally, you may still add even more `StripPrefix` filter
```
